rfcs/20210827-benchdnn-model-support/README.md

```diff
@@ -0,0 +1,88 @@
+integrate new features from oneDNN into framework to measure functionality
+designed for performance optimizations which increases risks not to deliver the
+targeted level of performance on time.
+
+It sounds like there could be some middle state between existing modes.
+Something that could run models with oneDNN but lacking dependencies on
+framework development and overheads. It may also provide better roofline
+projections with oneDNN and performance validation on framework side with
+numbers from this new mode should be simplified. Certain issues observed on
+frameworks side only would become easier to reproduce or confirm the problem is
+in a different place.
+
+## Motivation
+New mode targets oneDNN dev team and is supposed to cover two main directions:
+1) Ability to run primitives in environment __closer__ to framework. It may not
+   follow all the implementation details available on frameworks side, but if
+   it proves to reveal previously known issues it would mean that the flow is
+   viable, and it is reasonable to keep it.
+2) Ability for framework dev teams to provide standalone reproducers without
+   framework environment if statement (1) holds.
+
+This follows current benchdnn mission - to help developers find correctness and
+performance issues on a regular basis. It doesn't mean that no one else can use
+for some other purposes. Like now benchdnn is used for different kind of
+projections while it was not specifically designed for them.
+
+Another important notice is it doesn't have to cover the full model to be
+successful. Sometimes it would be enough to run two or three primitives
+consecutively to see the undesired performance effect. Better think about it
+as "chains" - several primitives run one by one once each. The ultimate and
+complete chain may be some existing model.
+
+## Proposal
+Let's introduce a new driver in benchdnn - "model". It would parse problems with
+their correspondent driver names, setup a problem, setup a primitive, manage
+memories, and pile them up. Once "model" driver is detected, all the further
+input will be parsed at once. This implies that only a single setting among
+supported and a single problem descriptor will be expected for each layer.
+Once input ends, a "chain" is executed in given order, layer by layer with a
+`wait` call in the very end.
+When parsing a problem, driver will create a primitive based on given settings
+and prepare memories the way that sources would be passed from previous layer
+destination.
+
+Pros:
+* Standalone and familiar tool using oneDNN only.
+* Same syntax to trigger __existing__ primitives and features.
+* Features available for other drivers would be transparently available in a
+  new mode, too.
+* Easier validation of performance gain for fusions - model mode would support
+  attributes and two runs with consecutive primitive as a standalone or as a
+  post-op would be supported.
```

Comment:
The wording confuses me here. Earlier it says "model mode would support attributes and"...then some other things, which we know are also supported because of the "and". So it seems like the last "would be supported" is just extra and unnecessary. It's also hard to understand what the second supported thing is--the two runs? A consecutive primitive? A post-op? You might be able to just delete the "would be supported" or insert "and also" in front of the other thing(s) that model mode supports. Adding a comma or two might also help clarify the meaning of the sentence. 

Here's a possible rewording: 

"...model mode would support attributes and also two runs with a consecutive primitive, as a standalone or a post-op.